viewer/packages/utilities/src/batchedDebounce.ts

```diff
@@ -66,13 +79,19 @@ export function batchedDebounce<T, R>(
     try {
       const results = await callback(inputs);
 
-      if (results.length !== inputs.length) {
-        throw new Error('Batch result length mismatch');
-      }
+      if (results === undefined) {
+        currentBatch.forEach(task => {
+          task.resolve(undefined as R);
```

Comment:
Would be nice with a small explanatory comment for why it's fine to cast `undefined` here 😅